rust-sdk/whirlpool/src/account.rs

```diff
@@ -0,0 +1,69 @@
+use serde::Deserialize;
+use serde_json::from_value;
+use solana_account_decoder::UiAccountData;
+use solana_client::{rpc_client::RpcClient, rpc_request::TokenAccountsFilter};
+use solana_program::pubkey::Pubkey;
+use solana_sdk::rent::Rent;
+use solana_sdk::sysvar::SysvarId;
+use std::{error::Error, str::FromStr};
+
+#[derive(Debug, Clone)]
+pub struct ParsedTokenAccount {
+    pub pubkey: Pubkey,
+    pub token_program: Pubkey,
+    pub mint: Pubkey,
+    pub amount: u64,
+}
+
+#[derive(Deserialize, Clone, Debug)]
+struct Parsed {
+    info: SplToken,
+}
+
+#[derive(Deserialize, Clone, Debug)]
+#[serde(rename_all = "camelCase")]
+struct SplToken {
+    mint: String,
+    token_amount: Amount,
+}
+
+#[derive(Deserialize, Clone, Debug)]
+#[serde(rename_all = "camelCase")]
+struct Amount {
+    amount: String,
+    ui_amount_string: String,
+    ui_amount: f64,
+    decimals: u8,
+}
+
+// This is a little hacky but it is done this way because
+// the original get_token_accounts_for_owner uses json_parsed encoding
+// and we don't want to use serde_json in this crate.
```

Comment:
nit: Now we can remove this comment because `get_token_accounts_by_owner` is used. 🙂